executor/src/witgen/mod.rs

```diff
@@ -102,9 +106,15 @@ where
     for row in 0..degree as DegreeType {
         // Check if we are in a loop.
         if looping_period.is_none() && row % 100 == 0 && row > 0 {
-            looping_period = rows_are_repeating(&values);
+            let relevant_values = values
+                .iter()
+                .enumerate()
+                .filter(|(id, _)| generator.is_relevant_witness(*id))
```

Comment:
Why was `is_relevant_witness` not used before?

Reply:
I guess strictly speaking that was a bug before, at least [Chis mentioned](https://github.com/powdr-org/powdr/pull/373#discussion_r1254805315) it wasn't intended. Before, all witness columns were used to detect loops, including those of other state machines. In practice, they are all 0 at this point, but that might change in the future.